src/native-state-wayland.cpp

```diff
@@ -368,3 +415,86 @@ NativeStateWayland::quit_handler(int /*signum*/)
     should_quit_ = true;
 }
 
+void
+NativeStateWayland::seat_handle_capabilities(void *data, struct wl_seat *seat, uint32_t caps)
+{
+    NativeStateWayland *that = static_cast<NativeStateWayland *>(data);
+
+    if ((caps & WL_SEAT_CAPABILITY_POINTER) && !that->display_->pointer) {
+        that->display_->pointer = wl_seat_get_pointer(seat);
+        wl_pointer_add_listener(that->display_->pointer, &pointer_listener_, that);
+    } else if (!(caps & WL_SEAT_CAPABILITY_POINTER) && that->display_->pointer) {
+        wl_pointer_destroy(that->display_->pointer);
+        that->display_->pointer = NULL;
+    }
+}
+
+
+void
+NativeStateWayland::pointer_handle_enter(void *data, struct wl_pointer *pointer,
+                                         uint32_t serial, struct wl_surface *surface,
+                                         wl_fixed_t sx, wl_fixed_t sy)
+{
+    NativeStateWayland *that = static_cast<NativeStateWayland *>(data);
+
+    struct wl_buffer *buffer;
+    struct my_cursor *my_cursor = that->cursor_;
+    struct wl_cursor *cursor;
+    struct wl_cursor_image *image;
+
+    cursor = my_cursor->default_cursor;
```

Comment:
With the current handling of cursors, my_cursor could be a nullptr.

Reply:
Added a check for it.